README.md

```diff
@@ -164,27 +161,42 @@ reading the following articles:
 - [Functional Swift for Dealing with Optional Values](http://robots.thoughtbot.com/functional-swift-for-dealing-with-optional-values)
 - [Railway Oriented Programming](http://fsharpforfunandprofit.com/posts/recipe-part2/)
 
+And check out this talk:
+
+- [How I Learned To Stop Worrying And Love The Functor](https://github.com/gfontenot/talks/tree/master/Functors)
+
 ## Usage
 
-The first thing you need to do when you receive JSON data is convert it to an
-instance of the `JSONValue` enum. This is done by passing the `AnyObject`
-value returned from `NSJSONSerialization` to `JSONValue.parse()`:
+The first thing you need to do when you receive JSON data is convert it from
+`NSData` to an `AnyObject` using the build in `NSJSONSerialization` API. Once
+you have the `AnyObject`, you can call the global `decode` function to get back
+the decoded model.
 
 ```swift
 let json: AnyObject? = NSJSONSerialization.JSONObjectWithData(responseData, options: NSJSONReadingOptions(0), error: nil)
 
 if let j: AnyObject = json {
-  let value: JSONValue = JSONValue.parse(j)
-  let user: User? = User.decode(value)
+  let user: User? = decode(j) // ignore error info or
+  let decodedUser: Decoded<User> = decode(j) // preserver error info
 }
 ```
 
 Note that you probably want to use an error pointer to track errors from
 `NSJSONSerialization`.
 
-The `JSONValue` enum exists to help with some of the type inference, and also
-wraps up some of the casting that we'll need to do to transform the JSON into
-native types.
+The `JSON` enum exists to help with some of the type inference, and also wraps
+up some of the casting that we'll need to do to transform the JSON into native
+types.
+
+With Argo 1.0, we've introduced a new type, `Decoded<T>`. This is now the type
+returned from the `decode` function that you implement as part of the
+`JSONDecodable` protocol. This new type allows you to preserver information
+about why a decoding failed. We can then use the global `decode` function to
+ignore that new type and just get back the optional value or keep the value
+contained within `Decoded` so you can inspect why the decoding failed. When you
+decode an `AnyObject` into a model using the global `decode` function, you can
+specify whether you want an `Optional` model or a `Decoded` model by specifying
+the return type as seen in the code block above.
 
 Next, you need to make sure that models that you wish to decode from JSON
 conform to the `JSONDecodable` protocol:
```

Comment:
We're mentioning JSONDecodable several times. If this PR is merged after the other one that renames that to Decodable, this will need to be changed.